fvm/evm/emulator/state/base_test.go

```diff
@@ -302,6 +302,164 @@ func TestBaseView(t *testing.T) {
 		require.Equal(t, uint64(1), view.NumberOfAccounts())
 	})
 
+	t.Run("test account iterator", func(t *testing.T) {
+		ledger := testutils.GetSimpleValueStore()
+		rootAddr := flow.Address{1, 2, 3, 4, 5, 6, 7, 8}
+		view, err := state.NewBaseView(ledger, rootAddr)
+		require.NoError(t, err)
+
+		accountCounts := 10
+		nonces := make(map[gethCommon.Address]uint64)
+		balances := make(map[gethCommon.Address]*uint256.Int)
+		codeHashes := make(map[gethCommon.Address]gethCommon.Hash)
+		for i := 0; i < accountCounts; i++ {
+			addr := testutils.RandomCommonAddress(t)
+			balance := testutils.RandomUint256Int(1000)
+			nonce := testutils.RandomBigInt(1000).Uint64()
+			code := testutils.RandomData(t)
+			codeHash := testutils.RandomCommonHash(t)
+
+			err = view.CreateAccount(addr, balance, nonce, code, codeHash)
+			require.NoError(t, err)
+
+			nonces[addr] = nonce
+			balances[addr] = balance
+			codeHashes[addr] = codeHash
+		}
+		err = view.Commit()
+		require.NoError(t, err)
+
+		ai, err := view.AccountIterator()
+		require.NoError(t, err)
+
+		counter := 0
+		for {
+			acc, err := ai.Next()
+			require.NoError(t, err)
+			if acc == nil {
+				break
+			}
+			require.Equal(t, nonces[acc.Address], acc.Nonce)
+			require.Equal(t, balances[acc.Address].Uint64(), acc.Balance.Uint64())
+			require.Equal(t, codeHashes[acc.Address], acc.CodeHash)
```

Comment:
The deletion ensures we see all unique account. Otherwise, returning the same account would also pass the tests.
```suggestion
			require.Equal(t, nonces[acc.Address], acc.Nonce)
			delete(nonces, acc.Address)
			require.Equal(t, balances[acc.Address].Uint64(), acc.Balance.Uint64())
			delete(balances, acc.Address)
			require.Equal(t, codeHashes[acc.Address], acc.CodeHash)
			delete(codeHashes, acc.Address)
```

Reply:
applied